### ISSUE TITLE
Russian Translation, Cyrillic Support

### ISSUE THREAD

**Issue by SvinkaSunshine** [cloudnode-pro/BankAccounts#121]
When using the MariaDB mode, translations/invoice descriptions containing Cyrillic characters are not saved. This issue is likely related to the database encoding.

<details>
<summary>Translation of config.yml</summary>

```
# Database configuration
db:
    # Select here the database you want to use
    # The following databases are supported:
    #  - sqlite - (default) stores all data in a local file
    #  - mariadb - allows using a remote database with JDBC
    # IMPORTANT! If you change the DB type, the data is not automatically migrated between databases.
    db: sqlite

    # SQLite
    sqlite:
        # Database file name (in the plugin's folder)
        # If the file does not exist, it will be created
        file: bank.db

    # MariaDB
    mariadb:
        # JDBC url
        jdbc: jdbc:mariadb://localhost:3306/your_db_name
        # Database user
        user: root
        # Database password
        password: ChangeThis!

    # Advanced configuration (do not touch unless you know what you are doing)
    # HikariCP. Comment out any data source property to disable setting it.
    cachePrepStmts: true
    prepStmtCacheSize: 250
    prepStmtCacheSqlLimit: 2048
    useServerPrepStmts: true
    useLocalSessionState: true
    rewriteBatchedStatements: true
    cacheResultSetMetadata: true
    cacheServerConfiguration: true
    elideSetAutoCommits: true
    maintainTimeStats: false

integrations:
    vault:
        # Enable Vault integration
        enabled: false

        # Transaction description for all vault operations
        description: Vault транзакция

currency:
    # Currency symbol
    symbol: $

    # Currency format
    # See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/text/DecimalFormat.html#special_pattern_character
    format: "#,##0.00"

# Starting balance
starting-balance: 0

# Server account
server-account:
    # Display name for the server account
    name: Central Bank
    # Account type
    # Possible values:
    #   0: Personal account
    #   1: Business account
    type: 1
    # Starting balance
    # Use `Infinity` for infinite ∞ balance
    starting-balance: Infinity

# Maximum number of accounts a player can have
# Account types:
#   0: Personal account
#   1: Business account
# Limits:
#   0: cannot have any accounts
#   -1: unlimited accounts
#
# Players can open accounts themselves if they have `bank.account.create`. These limits also apply if someone else
# creates bank accounts for other players with the admin permission `bank.account.create.other`. You can bypass the
# limits using `bank.account.create.bypass`.
account-limits:
    # Personal accounts
    0: 1
    # Business accounts
    1: -1

# Top balance leaderboard
# Accounts with infinite balance are excluded
baltop:
    # How many accounts to show per page
    per-page: 10

# Transfer confirmation
transfer-confirmation:
    # If enabled, players will be shown a transfer overview asking them to confirm the transfer
    enabled: true
    # Minimum transfer amount to show confirmation for
    # Min transfer amount is > 0, so setting this to 0 will enable confirmation for all transfers amounts
    min-amount: 200
    # If true, confirmation will not be required for transfers between own accounts
    bypass-own-accounts: true

# Transactions history
history:
    # Number of transactions to return per page
    # You can use `/bank transactions <account> --all` to view all transactions
    per-page: 10

# Payment instruments
# These are like bank cards that can be used to make payments.
instruments:
    # The in-game item to use. See https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html
    material: PAPER

    # Require the player to have an item in their inventory of the same material in order to create a new instrument.
    # The item is consumed and replaced with the instrument. This is recommended to prevent limitless generation of
    # items. Payment instruments CAN be used in crafting recipes and villager trading as if they were regular items.
    # You can bypass this requirement using `bank.instrument.create.bypass`. Non-players (e.g. console, command blocks,
    # etc.) always bypass this requirement.
    require-item: true

    # The name of the item
    # Placeholders:
    #  - <account> - Account name. Defaults to owner username or ID if not set.
    #  - <account-id> - Account ID
    #  - <account-type> - Account type (Personal or Business)
    #  - <account-owner> - Account owner username
    #  - <date> - Date card was created. You can use <date:'yyyy-MM-dd HH:mm:ss'>. Check here for patterns: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html#patterns
    # NOTE: The placeholders are set only when the item is created.
    # If the account is renamed or the balance changes, the text on the item will NOT be updated.
    # Additionally, if you modify this, it will only be effective for new bank cards.
    name: "<green><b>Банковская карта</b></green> <aqua>—</aqua> <gray><account></gray>"

    # Lore of the item (text below the name)
    # Same placeholders as name
    lore:
        - "<white><account-type>, <account></white>"
        - "<gray>#<account-id></gray>"
        - "<gray>Дата создания: <date:'MM/yy'></gray>"
        - "<gray>Владелец:</gray> <white><account-owner></white>"

    # Item glint
    # This makes the item shine as if enchanted
    glint:
        enabled: true

        # The hidden enchantment to apply to the item
        # Enchantments are the same as seen in the `/enchant` command. Do not include the `minecraft:` prefix.
        # If you set this to something like "vanishing_curse" the item will disappear when the player dies.
        # If you use a compatible enchantment with the item material, the enchantment will be fully functional.
        # (e.g. if your item is a sword, and you use "sharpness", the sword will deal more damage, even though the
        # enchantment is not listed in the lore. it is recommended to avoid that)
        # Level 1 is always applied.
        enchantment: unbreaking

# POS
# Note: you cannot use <hover> <click> etc. in GUIs
pos:
    # Allow creating POS with a PERSONAL account
    # If you want to require players to use a business account, set this to false.
    # You can bypass this with `bank.pos.create.personal`
    allow-personal: false

    # The title shown at the top of the GUI when opening a POS
    # Placeholders:
    # <description> - POS description
    # <price> - Price of the POS contents without formatting, example: 123456.78
    # <price-formatted> - Price of the POS contents with formatting, example: $123,456.78
    # <price-short> - Price of the POS contents with formatting, example: $123k
    title: "<green><bold>POS</bold></green> <gray>— <price-formatted></gray>"

    # POS info item
    info:
        # Available materials: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html
        material: PAPER

        # Glint / glow (does not work on all items)
        glint: true

        # Item name when owner is viewing the POS.
        # Same placeholders as POS title.
        name-owner: "<green><bold>POS</bold></green>"
        # Item name when potential buyer is viewing the POS.
        # Same placeholders as POS title.
        # Extra placeholders: same as `messages.balance`. The account is of the owner of the POS.
        name-buyer: "<green><bold>POS</bold></green> <gray>— <account></gray>"

        # Lore lines when owner is viewing the POS.
        # Same placeholders as POS title.
        lore-owner:
            - "<gray>Цена:</gray> <green><price-formatted></green>"
            - "<gray>Описание:</gray> <white><description></white>"
        # Same placeholders as name-buyer
        lore-buyer:
            - "<gray>Цена:</gray> <green><price-formatted></green>"
            - "<gray>Описание:</gray> <white><description></white>"

    # POS delete item (owner only)
    # No placeholders.
    delete:
        material: BARRIER
        glint: false
        name: "<red><bold>Удалить POS</bold></red>"
        lore:
            - "<gray>Нажмите, чтобы удалить этот POS.</gray>"

    # Confirm purchase item (buyer only)
    # Same placeholders as name-buyer, but this time the account is of the buyer.
    confirm:
        material: LIME_STAINED_GLASS_PANE
        glint: false
        name: "<green><bold>Подтвердите покупку</bold></green>"
        lore:
            - "<gray>Нажмите, чтобы подтвердить покупку.</gray>"
            - "<gray>Цена:</gray> <green><price-formatted></green>"
            - "<gray>Баланс:</gray> <white><balance-formatted></white>"
            - "<gray>Оплата со счета:</gray> <white><account></white> <gray>(<account-type>)</gray>"
            - "<dark_gray>Аккаунт ID <account-id></dark_gray>"
    # Deny purchase item (buyer only)
    # No placeholders.
    decline:
        material: RED_STAINED_GLASS_PANE
        glint: false
        name: "<red><bold>Отклонить покупку</bold></red>"
        lore:
            - "<gray>Нажмите, чтобы отказаться от этой покупки.</gray>"

# Interest rate
# Interest is paid from/to the server account.
# If the server account has insufficient funds, interest payments will silently fail.
# NOTE: A transaction is created for each interest payment.
interest:
    # Personal accounts
    0:
        # Percent to add to the balance. If negative, it will be subtracted. Set to 0 to disable interest.
        # Acceptable range: -100–100
        rate: 0
        # Interval time in real-life MINUTES for how often interest is added.
        # Default: 24 hours (i.e. midnight every day).
        interval: 1440
        # Description shown in the transaction history
        # Available placeholders:
        # - <rate> - Interest rate as decimal (e.g. "5")
        # - <rate-formatted> - Formatted interest rate (e.g. "5%")
        # - <balance> - Account balance before interest (e.g. "123456.78")
        # - <balance-formatted> - Formatted balance before interest (e.g. "$123,456.78")
        # - <balance-short> - Formatted balance before interest (e.g. "$123k")
        description: "Выплата процентов (<rate-formatted>)"
    # Business accounts
    1:
        rate: 0
        interval: 1440
        description: "Выплата процентов (<rate-formatted>)"

# Invoice
invoice:
    # Number of invoices to return per page
    per-page: 10

# Messages
messages:
    # Command usage message
    # Placeholders:
    # - <command> - The command name (e.g. "bank")
    # - <arguments> - The command usage arguments (e.g. "send <from> <to> <amount> [description]")
    command-usage: "<yellow>(!) Используйте:</yellow> <white>/<command> <arguments></white>"

    # Account types
    types:
        # Personal account
        0: Personal
        # Business account
        1: Business
        # Vault integration account used to integrate payments with other plugins
        2: Checking

    # Errors
    errors:
        # You have no accounts
        no-accounts: "<red>(!) У вас нет банковских счетов.</red>"
        # No permission
        no-permission: "<red>(!) У вас нет разрешения на использование этой команды.</red>"
        # Account not found
        account-not-found: "<red>(!) Аккаунт не найден.</red>"
        # Command not recognised
        unknown-command: "<red>(!) Неизвестная команда. Используйте <click:suggest_command:/<label> help><gray>/<label> help</gray></click>.</red>"
        # <type> the account type
        # <limit> the account limit for that type
        max-accounts: "<red>(!) Вы не можете иметь более <limit> <type> аккаунтов.</red>"
        rename-vault-account: "<red>(!) Вы не можете переименовать этот аккаунт.</red>"
        # You do not own this account
        not-account-owner: "<red>(!) Вы не владелец этого аккаунта.</red>"
        # Account frozen (placeholders same as balance)
        frozen: "<gold>(!) <account-type> аккаунт <gray><account> (<account-id>)</gray> заморожен. Этот аккаунт не может инициировать или получать транзакции.</gold>"
        # Cannot transfer to same account
        same-from-to: "<red>(!) Счета «от» и «для» должны быть разными.</red>"
        # You can only make transfer between your own accounts
        transfer-self-only: "<red>(!) Вы можете осуществлять переводы только между своими счетами.</red>"
        # You can only transfer to account that are not yours
        transfer-other-only: "<red>(!) Вы не можете осуществлять перевод между своими счетами.</red>"
        # Invalid number. Placeholder <number> is the invalid number
        invalid-number: "<red>(!) Неправильный номер: <gray><number></gray></red>"
        # $0 or negative transfer
        negative-transfer: "<red>(!) Сумма перевода должна быть больше нуля.</red>"
        # Insufficient funds (placeholders same as balance)
        insufficient-funds: "<red>(!) На вашем счету недостаточно средств. У вас есть <gray><balance-formatted></gray></red>"
        # Trying to close an account that has non-zero balance
        closing-balance: "<red>(!) Вы не можете закрыть счет, баланс которого не равен нулю. На этом аккаунте <gray><balance-formatted></gray></red>"
        # Command only for players
        player-only: "<red>(!) Вы должны быть игроком, чтобы использовать эту команду.</red>"
        # Player not found
        player-not-found: "<red>(!) Игрок не найден.</red>"
        # You must have a specific item to convert to an instrument
        # Placeholders:
        # <material> - the item type that is required (the item ID)
        # <material-key> - item translation key, can be used together with <lang:<material-key>>
        instrument-requires-item: "<red>(!) Вы должны иметь <gray>x1 <lang:<material-key>></gray> в вашем инвентаре, чтобы создать банковскую карту.</red>"
        # Target player's inventory is full
        # Placeholder: <player> - the target player's name
        target-inventory-full: "<red>(!) Инвентарь игрока <gray><player></gray> полон.</red>"
        block-too-far: "<red>(!) Целевой блок находится слишком далеко.</red>"
        pos-already-exists: "<red>(!) POS уже существует по этому адресу.</red>"
        pos-not-chest: "<red>(!) Целевой блок не является сундуком.</red>"
        pos-double-chest: "<red>(!) POS не может быть двойным сундуком.</red>"
        pos-empty: "<red>(!) Сундук пуст. POS отменен.</red>"
        pos-invalid-card: "<red>(!) Для использования POS вам нужна действующая банковская карта.</red>"
        pos-no-permission: "<red>(!) У вас нет разрешения на использование POS.</red>"
        no-card: "<red>(!) Чтобы использовать эту услугу, вы должны держать свою банковскую карту.</red>"
        pos-items-changed: "<red>(!) Предметы в сундуке изменились. POS-терминал отменен.</red>"
        pos-create-business-only: "<red>(!) Вы можете создать POS только с бизнес-аккаунтом.</red>"
        # Provided string includes disallowed characters
        # Placeholder: <characters> - the disallowed characters
        disallowed-characters: "<red>(!) Предоставленная строка содержит недопустимые символы: <gray><characters></gray></red>"
        # Account is already frozen (placeholders same as balance)
        already-frozen: "<yellow>(!) Этот аккаунт уже заморожен.</yellow>"
        # Account is not frozen (placeholders same as balance)
        not-frozen: "<gold>(!) Этот аккаунт не заморожен.</gold>"
        # Invoice negative amount
        negative-invoice: "<red>(!) Сумма счета должна быть больше нуля.</red>"
        # Invoice not found
        invoice-not-found: "<red>(!) Счет не найден.</red>"
        # Invoice seller account is the same as payer
        invoice-pay-self: "<red>(!) Вы не можете оплатить свой собственный счет с того же аккаунта. Вы можете использовать другую учетную запись.</red>"
        invoice-already-paid: "<red>(!) Этот счет уже оплачен.</red>"
        # Cannot send this invoice to this player because either:
        # a) the invoice has a set buy and that's not this player; OR
        # b) the player does not have permission to view invoices not meant for them
        invoice-cannot-send: "<red>(!) Вы не можете отправить этот счет этому игроку, поскольку у него нет разрешения на его просмотр.</red>"
        # Player has never played on this server
        player-never-joined: "<red>(!) Этот игрок никогда не присоединялся к этому серверу.</red>"
        # Asynchronous code failed. Detailed info is outputted in the console
        async-failed: "<red>(!) Ошибка запроса. Подробности смотрите в консоли.</red>"
        # Trying to delete vault integration account
        delete-vault-account: "<red>(!) Вы не можете удалить эту учетную запись.</red>"

    # Account balance
    # Available placeholders:
    # <account> - Account name. Defaults to owner username or ID if not set.
    # <account-id> - Account ID
    # <account-type> - Account type (Personal or Business)
    # <account-owner> - Account owner username
    # <balance> - Account balance without formatting, example: 123456.78
    # <balance-formatted> - Account balance with formatting, example: $123,456.78
    # <balance-short> - Account balance with formatting, example: $123k
    balance: |-
        <green><account></green> <gray>(<account-type> account)</gray>
        <green>Баланс:</green> <white><balance-formatted></white>
        <white><i>Владелец: <account-owner></i></white>
        <i><gray><click:copy_to_clipboard:<account-id>><hover:show_text:'Нажмите, чтобы скопировать аккаунт ID'><account-id></hover></click></gray></i>

    # List accounts
    list-accounts:
        header: "<green>Банковские аккаунты</green>"
        # Same placeholders as balance
        entry: "<dark_gray>*</dark_gray> <gray><click:run_command:/bank balance <account-id>><hover:show_text:'<white><account></white><newline><gray><account-type><dark_gray>:</dark_gray> <green><balance-formatted></green></gray><newline><gray><account-id></gray><newline><gray>Нажмите, чтобы просмотреть'><account></hover></click><dark_gray>:</dark_gray> <white><balance-short></white></gray>"

    # Plugin reload
    reload: "<green>(!) Конфигурация плагина перезагружена.</green>"

    # Account created (same placeholders as balance)
    account-created: "<green>(!) Успешно создан <account-type> аккаунт: <account></green> <click:run_command:/bank balance <account-id>><gray>Нажмите, чтобы просмотреть</gray></click>"

    # Balance set (same placeholders as balance)
    balance-set: "<green>(!) Успешно установлен <white><account></white> <gray>(<account-type>, <account-id>)</gray> баланс <white><balance-formatted></white></green>"

    # Account name set (same placeholders as balance)
    name-set: "<green>(!) Аккаунт </green><white><account></white> <green>успешно переименован."

    # Account has been successfully frozen (same placeholders as balance)
    account-frozen: "<aqua>(!) Аккаунт </aqua><white><account></white><aqua> успешно заморожен."

    # Account has been successfully unfrozen (same placeholders as balance)
    account-unfrozen: "<green>(!) Счет </green><white><account></white><green> успешно разморожен."

    # Account deleted (same placeholders as balance)
    account-deleted: "<green>(!) Аккаунт </green><white><account></white><green> успешно удален."

    # Transfer confirmation
    # Placeholders are the same as balance. Add prefix `from-` for account sending and `to-` for account receiving
    # Example: <to-account-id>, <from-balance-formatted>
    # Additional placeholders:
    # <amount> - Transfer amount without formatting, example: 123456.78
    # <amount-formatted> - Transfer amount with formatting, example: 123,456.78
    # <amount-short> - Transfer amount with formatting, example: 123k
    # <description> - Transfer description
    # <confirm-command> - Command to run to confirm transfer
    confirm-transfer: |-
        <green><bold>ПОДТВЕРДИТЬ ПЕРЕВОД</bold></green>
        <green>От:</green> <white><from-account></white> <gray>(<from-account-type> аккаунт <from-account-owner>, <from-account-id>)</gray>
        <green>Для:</green> <white><to-account></white> <gray>(<to-account-type> аккаунт <to-account-owner>, <to-account-id>)</gray>
        <green>Сумма:</green> <white><amount-formatted></white>
        <green>Описание:</green> <white><description></white>
        
        <click:run_command:<confirm-command>><hover:show_text:'<green>Нажмите, чтобы подтвердить перевод</green>'><green>[Я ПОДТВЕРЖДАЮ]</green></hover></click>

    # Transfer sent (same placeholders as confirm-transfer, except confirm-command)
    # Additional placeholders:
    # <transaction-id> - Transaction ID
    transfer-sent: <green>(!) Успешно отправлено <white><amount-formatted></white> от <hover:show_text:'<white><from-account></white><newline><gray><from-account-type><dark_gray>:</dark_gray> <green><from-balance-formatted></green></gray><newline><gray>Владелец <from-account-owner></gray><newline><gray><from-account-id></gray>'><white><from-account></white></hover> к <hover:show_text:'<white><to-account></white><newline><gray><to-account-type></gray><newline><gray>Владелец <to-account-owner></gray><newline><gray><to-account-id></gray>'><white><to-account></white></hover></green> <click:copy_to_clipboard:<transaction-id>><hover:show_text:'Нажмите, чтобы скопировать ID транзакции'><gray>#<transaction-id></gray></hover></click>

    # Transfer received (same placeholders as transfer-sent)
    transfer-received: <green>(!) Вы получили <white><amount-formatted></white> от <hover:show_text:'<white><from-account></white><newline><gray><from-account-type></gray><newline><gray>Владелец <from-account-owner></gray><newline><gray><from-account-id></gray>'><white><from-account></white></hover> в ваш <hover:show_text:'<white><to-account></white><newline><gray><to-account-type><dark_gray>:</dark_gray> <green><to-balance-formatted></green></gray><newline><gray>Владелец <to-account-owner></gray><newline><gray><to-account-id></gray>'><white><to-account></white></hover> аккаунт:</green> <white><description></white> <click:copy_to_clipboard:<transaction-id>><hover:show_text:'Нажмите, чтобы скопировать ID транзакции '><gray>(#<transaction-id>)</gray></hover></click>

    # Transaction history
    history:
        # All placeholders of `balance` are available
        # Additional placeholders:
        # <page> - Current page number
        # <max-page> - Maximum number of pages
        # <cmd-prev> - Command to go to previous page
        # <cmd-next> - Command to go to next page
        header: "<green>История переводов </green> <gray>(<hover:show_text:'<white><account></white><newline><gray><account-type><dark_gray>:</dark_gray> <green><balance-formatted></green></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><account></hover>)</gray>"
        # All placeholders of `balance` and `transfer-sent` are available. Use <account> (no prefix) for your account and "other-" prefix for other account
        # Additional placeholders:
        # <instrument> - The way the transfer was made, e.g. "direct transfer", "pos", etc.
        # <date> - Transaction date. You can use <date:'yyyy-MM-dd HH:mm:ss'>. Check here for patterns: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html#patterns
        # <full_date> - Full date with time zone. Can be used in <hover>, <click> etc.
        entry: "<hover:show_text:'<full_date>'><gray>#<transaction-id></gray></hover> <#4ade80><amount-formatted></#4ade80> <white><hover:show_text:'<white><other-account></white><newline><gray><other-account-type></gray><newline><gray>Владелец <other-account-owner></gray><newline><gray><other-account-id></gray>'><other-account></hover>: <description></white> <gray>(<instrument>)</gray>"
        # Same placeholders as header
        footer: "<click:run_command:<cmd-prev>><hover:show_text:'<white>Предыдущая страница</white>'><green>← Назад</green></hover></click> <gray>Страница <page>/<max-page></gray> <click:run_command:<cmd-next>><hover:show_text:'<white>Следущая страница</white>'><green>Вперёд →</green></hover></click>"
        # No transactions
        no-transactions: "<gray>(!) Нет транзакций для отображения.</gray>"

    # Successfully created payment instrument (bank card)
    instrument-created: "<green>(!) Платежный инструмент успешно создан.</green>"

    # Pos created
    # Same placeholders as balance (account is that of the seller).
    # Additional placeholders:
    # <price> - Price of the POS contents without formatting, example: 123456.78
    # <price-formatted> - Price of the POS contents with formatting, example: 123,456.78
    # <price-short> - Price of the POS contents with formatting, example: 123k
    # <description> - Bank statement description
    # <x> - X coordinate of the chest
    # <y> - Y coordinate of the chest
    # <z> - Z coordinate of the chest
    # <world> - World name of the chest
    pos-created: "<green>(!) POS-терминал с ценой </green><white><price-formatted></white><green> успешно создан. Получатель средств: <hover:show_text:'<white><account></white><newline><gray><account-type><dark_gray>:</dark_gray> <green><balance-formatted></green></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><white><account></white></hover>:</green> <gray><description></gray>"

    # Pos removed
    # (no placeholders)
    pos-removed: "<gold>(!) POS удален.</gold>"

    # Pos purchase completed
    # Same placeholders as transfer-sent. `from` account is buyer, `to` account is seller.
    # Additional placeholders:
    # <items> - Number of items purchased
    # <items-formatted> - E.g. `1 item`, `5 items`, etc.
    # <pos> - POS coordinates, e.g. `X: 123 Y: -12 Z: 456 in world`
    pos-purchase: "<green>(!) Успешно куплено <white><items-formatted></white> у <hover:show_text:'<white><to-account></white><newline><gray><to-account-type></gray><newline><gray>Владелец <to-account-owner></gray><newline><gray><to-account-id></gray>'><white><to-account></white></hover> за <white><amount-formatted></white>. Вы заплатили со счета <hover:show_text:'<white><from-account></white><newline><gray><from-account-type><dark_gray>:</dark_gray> <green><from-balance-formatted></green></gray><newline><gray>Владелец <from-account-owner></gray><newline><gray><from-account-id></gray>'><white><from-account></white></hover>:</green> <gray><description></gray>"

    # Someone made a purchase from your POS
    # Same placeholders as pos-purchase. `from` account is buyer, `to` account is seller.
    # Additional placeholders:
    # <items> - Number of items purchased
    # <items-formatted> - E.g. `1 item`, `5 items`, etc.
    # <pos> - POS coordinates, e.g. `X: 123 Y: -12 Z: 456 in world`
    pos-purchase-seller: "<green>(!) Продано <white><items-formatted></white> для <hover:show_text:'<white><from-account></white><newline><gray><from-account-type></gray><newline><gray>Владелец <from-account-owner></gray><newline><gray><from-account-id></gray>'><white><from-account></white></hover> за <white><amount-formatted></white> в POS <gray><pos></gray>. Средства пришли на счет<hover:show_text:'<white><to-account></white><newline><gray><to-account-type><dark_gray>:</dark_gray> <green><to-balance-formatted></green></gray><newline><gray>Владелец <to-account-owner></gray><newline><gray><to-account-id></gray>'><white><to-account></white></hover>:</green> <gray><description></gray>"

    # Account whois
    # Same placeholders as balance
    whois: |-
        <green>Аккаунт</green> <white><account></white>
        <gray>Владелец:</gray> <white><account-owner></white>
        <gray>Тип:</gray> <white><account-type></white>
        <gray>ID: <click:copy_to_clipboard:<account-id>><hover:show_text:'<white>Скопировать в буфер обмена</white>'><white><account-id></white></hover></click></gray>

    # Top balances
    baltop:
        # Baltop header
        # Placeholders:
        # <category> - Baltop category
        # <page> - Current page number
        # <cmd-prev> - Command to go to previous page
        # <cmd-next> - Command to go to next page
        header: "<green><bold>Балтоп</bold></green> <gray>(<category>)</gray><newline><click:run_command:<cmd-prev>><hover:show_text:'<white>Нажмите</white>'><green>← Назад</green></hover></click>     <gray>Страница <page></gray>        <click:run_command:<cmd-next>><hover:show_text:'<white>Нажмите</white>'><green>Вперёд →</green></hover></click><newline><dark_gray><st>                                            </st></dark_gray>"
        # Baltop entry
        # All placeholders from balance
        # <position> - leaderboard rank number
        entry: "<hover:show_text:'<white><account></white><newline><gray><account-type><dark_gray>:</dark_gray> <green><balance-formatted></green></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><gray>#<position></gray> <white><balance-short></white> <green><account></green></hover>"
        # Baltop player entry
        # Placeholders:
        # <position> - leaderboard rank number
        # <uuid> - Player UUID
        # <username> - Player username
        # <balance> - Total balance without formatting, example: 123456.78
        # <balance-formatted> - Total balance with formatting, example: $123,456.78
        # <balance-short> - Total balance with formatting, example: $123k
        entry-player: "<click:suggest_command:/msg <username> ><hover:show_text:'<white><username></white><newline><gray><balance-formatted></gray><newline>'><gray>#<position></gray> <white><balance-short></white> <green><username></green></hover></click>"

    # Invoices
    invoice:
        # Viewing an invoice
        # Placeholders:
        # <invoice-id> - Invoice ID
        # <amount> - Invoice amount without formatting, example: 123456.78
        # <amount-formatted> - Invoice amount with formatting, example: $123,456.78
        # <amount-short> - Invoice amount with formatting, example: $123k
        # <description> - Invoice description
        # <date> - Date invoice was created. You can use <date:'yyyy-MM-dd HH:mm:ss'>. Check here for patterns: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html#patterns
        # <buyer> - The username of the player who is requested to pay the invoice. Can be "anyone" if anyone can pay this invoice
        # <transaction-id> - Invoice transaction ID (if the invoice is paid). If not paid, will be "unpaid"
        # <status> - Payment status (as defined below)
        # <pay-button> - Button to pay the invoice (as defined below)
        # <account> - Seller account name. Defaults to owner username or ID if not set.
        # <account-id> - Seller account ID
        # <account-type> - Seller account type (Personal or Business)
        # <account-owner> - Seller account owner username
        # <balance> - Seller account balance without formatting, example: 123456.78
        # <balance-formatted> - Seller account balance with formatting, example: $123,456.78
        # <balance-short> - Seller account balance with formatting, example: $123k
        details: |-
            <bold><green>СЧЕТ</green></bold> <gray><hover:show_text:'<white>Нажмите, чтобы скопировать</white>'><click:copy_to_clipboard:<invoice-id>>#<invoice-id></click></hover></gray>
            <white>Сумма: <green><amount-formatted></green></white>
            <white>От: <hover:show_text:'<white><account></white><newline><gray><account-type></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><gray><account></gray></hover>
            <white>Для: <gray><buyer></gray></white>
            <white>Дата: <gray><date:'yyyy-MM-dd HH:mm'> UTC</gray></white>
            <white>Статус:</white> <status>
            <white>Описание: <gray><description></gray></white><pay-button>

        # Used in the <status> placeholder
        # Same placeholders as details (except <status>)
        status:
            paid: "<green>Оплачен</green> <gray>(transaction #<transaction-id>)</gray>"
            unpaid: <gray>Неоплачен</gray>

        # Used in the <pay-button> placeholder. This placeholder is an empty string if the invoice is unpaid.
        # Same placeholders as details (except <pay-button>)
        pay-button: <newline><newline><hover:show_text:'<white>Click to pay</white>'><click:suggest_command:/invoice pay <invoice-id>><bold><green>[Оплатить]</green></bold></click></hover>

        # You have received an invoice
        # Same placeholders as details
        received: <click:run_command:/invoice view <invoice-id>><green>(!)Вы получили счет <hover:show_text:'<green>Счет</green> <gray>#<invoice-id></gray><newline><gray>От <white><account></white></gray><newline><gray>  <white><amount-formatted></white></gray>'><gray>#<invoice-id></gray></hover> для <hover:show_text:'<white><account></white><newline><gray><account-type></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><gray><account></gray></hover>. Нажмите, чтобы просмотреть.</green></click>

        # Invoice successfully sent
        # Same placeholders as details
        sent: <green>(!) Счет <gray>#<invoice-id></gray> успешно отправлен.</green>

        # Invoice created
        # Same placeholders as details
        created: <click:run_command:/invoice view <invoice-id>><green>(!) Счет-фактура <gray>#<invoice-id></gray> успешно создана на сумму <gray><amount-formatted></gray> для <gray><buyer></gray>, <hover:show_text:'<white><account></white><newline><gray><account-type><dark_gray>:</dark_gray> <green><balance-formatted></green></gray><newline><gray><account-id></gray>'><gray><account></gray></hover>. Нажмите, чтобы просмотреть.</green></click>

        # Your invoice has been paid (seller's perspective)
        # Same placeholders as details
        paid-seller: <click:run_command:/invoice view <invoice-id>><green>(!) Ваш счет <hover:show_text:'<green>Invoice</green> <gray>#<invoice-id></gray><newline><gray>на <white><amount-formatted></white></gray>'><gray>#<invoice-id></gray></hover> <gray>(<amount-short>)</gray> был оплачен. Нажмите, чтобы просмотреть.</green></click>

        # You have successfully paid an invoice (buyer's perspective)
        # Same placeholders as details
        paid-buyer: <click:run_command:/invoice view <invoice-id>><green>(!) Вы успешно оплатили счет <hover:show_text:'<green>Счет</green> <gray>#<invoice-id></gray><newline><gray>От <white><account></white></gray><newline><gray>на <white><amount-formatted></white></gray>'><gray>#<invoice-id></gray></hover> <gray>(<amount-short>)</gray>. Нажмите, чтобы просмотреть.</green></click>

        # List invoices
        list:
            # Header
            # Placeholders:
            # <page> - current page number
            # <cmd-prev> - command to go to previous page
            # <cmd-next> - command to go to next page
            header: "<green>Счета-фактуры</green>"

            # Invoice entry
            # Same placeholders as details
            entry: <click:run_command:/invoice view <invoice-id>><dark_gray>#<invoice-id></dark_gray> <white><amount-formatted></white> <gray><status>,</gray> <hover:show_text:'<white><account></white><newline><gray><account-type></gray><newline><gray>Владелец <account-owner></gray><newline><gray><account-id></gray>'><gray><account></gray></hover></click>

            # Footer
            # Same placeholders as details
            footer: "<hover:show_text:'Нажмите, чтобы увидеть предыдущую страницу'><click:run_command:<cmd-prev>><green>← Назадs</green></click></hover> <gray>Страница <page></gray> <hover:show_text:'Нажмите, чтобы увидеть следующую страницу'><click:run_command:<cmd-next>><green>Вперёд →</green></click></hover>"

    # New version available
    # Placeholders:
    # <version> - New version
    update-available: "<hover:show_text:'<white>Нажмите, чтобы просмотреть обновление</white><newline><gray> - Журнал изменений и примечания к выпуску<newline> - Ссылки для скачивания'><click:open_url:https://modrinth.com/plugin/bankaccounts/version/<version>><dark_gray>[<green>BankAccounts</green>]</dark_gray> <white>Новая версия <green><version></green> выпущена.</white><newline><gray>Пожалуйста, обновите, чтобы получить последние исправления ошибок и функции.</gray><newline> <green>> Нажмите, чтобы просмотреть обновление<green></click></hover>"
```
</details>

**Comment by zefir-git** [comment]
You are right, this is most likely due to using the latin1 character set for MariaDB. This character set supports just Latin A-z, numbers and punctuation. Table initialisation needs to be updated with a more permissive character set.

Until this is fixed, to avoid this bug, please only use Latin letters for account names, transaction descriptions, POS descriptions and invoice descriptions.

- Account name:
  https://github.com/cloudnode-pro/BankAccounts/blob/c540a90f209be9044db7aae612c035756ce30f64/src/main/resources/db-init/mysql.sql#L6

- Transaction description:
  https://github.com/cloudnode-pro/BankAccounts/blob/c540a90f209be9044db7aae612c035756ce30f64/src/main/resources/db-init/mysql.sql#L19

- POS description:
  https://github.com/cloudnode-pro/BankAccounts/blob/c540a90f209be9044db7aae612c035756ce30f64/src/main/resources/db-init/mysql.sql#L31

- Invoice description:
  https://github.com/cloudnode-pro/BankAccounts/blob/c540a90f209be9044db7aae612c035756ce30f64/src/main/resources/db-init/mysql.sql#L45

**Comment by zefir-git** [comment]
The fix for this (#122) has been released in 1.8.2
